client/verta/verta/registry/entities/_modelversion.py

```diff
@@ -1514,3 +1519,95 @@ def set_hide_output_label(self, hide):
     def get_hide_output_label(self):
         self._refresh_cache()
         return self._msg.hide_output_label
+
+    def log_dataset_version(self, key, dataset_version):
+        """
+        Logs a Verta DatasetVersion to this Model Version with the given key.
+
+        Parameters
+        ----------
+        key : str
+            Name of the dataset version.
+        dataset_version : :class:`~verta.dataset.entities.DatasetVersion`
+            Dataset version.
+
+        """
+        if not isinstance(dataset_version, _dataset_version.DatasetVersion):
+            raise TypeError("`dataset_version` must be of type DatasetVersion")
+
+        artifact_msg = _CommonCommonService.Artifact(
+            key=key,
+            path_only=True,
+            artifact_type=_CommonCommonService.ArtifactTypeEnum.DATA,
+            linked_artifact_id=dataset_version.id,
+        )
+
+        msg = _RegistryService.LogDatasetsInModelVersion(
+            model_version_id=self.id,
+            datasets=[artifact_msg],
+        )
+
+        endpoint = "/api/v1/registry/model_versions/{}/logDatasets".format(
+            self.id,
+        )
+
+        response = self._conn.make_proto_request("POST", endpoint, body=msg)
+        self._conn.must_response(response)
+        self._clear_cache()
+
+    def get_dataset_version(self, key):
+        """
+        Gets the DatasetVersion with name `key` from this Model Version.
+
+        Parameters
+        ----------
+        key : str
+            Name of the dataset version.
+
+        Returns
+        -------
+        :class:`~verta.dataset.entities.DatasetVersion`
+            DatasetVersion associated with the given key.
+
+        """
```

Comment:
I forgot to mention! We should refresh the cache like this on any `get_` to get `self._msg` from registry backend

```suggestion
        """
        self._refresh_cache()
```

Reply:
Also, this change means we will not _need_ the multiple

```python
model_version = client.get_registered_model_version(id=model_version.id)
```

in the test.